src/DataBoxEdge/DataBoxEdge/Common/Cmdlets/Triggers/DataBoxEdgeTriggerNewCmdletBase.cs

```diff
@@ -0,0 +1,250 @@
+        [ValidateNotNullOrEmpty]
+        public string ShareId { get; set; }
+
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventSwitchParameter)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerResourceIdParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventSwitchParameter)]
+        [ValidateNotNullOrEmpty]
+        public SwitchParameter PeriodicTimerEvent { get; set; }
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventScheduleHelpMessage)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerResourceIdParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventScheduleHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string Schedule;
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventStartTimeHelpMessage)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerResourceIdParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventStartTimeHelpMessage)]
+        public DateTime StartTime;
+
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventTopicHelpMessage)]
+        [Parameter(Mandatory = true,
+            ParameterSetName = PeriodicTimerTriggerResourceIdParameterSet,
+            HelpMessage = HelpMessageTrigger.PeriodicTimerEventTopicHelpMessage)]
+        public string Topic;
+
+        [Parameter(Mandatory = false, HelpMessage = Constants.AsJobHelpMessage)]
+        public SwitchParameter AsJob { get; set; }
+
+        private ResourceModel _trigger;
+
+        private ResourceModel GetResourceModel()
+        {
+            return TriggersOperationsExtensions.Get(
+                this.DataBoxEdgeManagementClient.Triggers,
+                this.DeviceName,
+                this.Name,
+                this.ResourceGroupName);
+        }
+
+        private string GetRoleId()
+        {
+            return RolesOperationsExtensions.Get(
+                this.DataBoxEdgeManagementClient.Roles,
+                this.DeviceName,
+                this.RoleName,
+                this.ResourceGroupName).Id;
+        }
+
+        private string GetShareId()
+        {
+            return SharesOperationsExtensions.Get(
+                this.DataBoxEdgeManagementClient.Shares,
+                this.DeviceName,
+                this.ShareName,
+                this.ResourceGroupName).Id;
+        }
+
+        private string GetResourceFoundMessage()
```

Comment:
It's no need to make it a function if it only used once.

Reply:
Since this is a generated message from constants, Just made it to keep the other function clean